lib/presentation/mixins/handle_video_download_mixin.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:io';
+
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/widgets/video_viewer_dialog.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:universal_html/html.dart' as html;
+
+mixin HandleVideoDownloadMixin {
+  Future<String> handleDownloadVideoEvent(
+    BuildContext context,
+    Event event,
+  ) async {
+    lastSelectedVideoEventId = event.eventId;
+    final videoFile = await event.downloadAndDecryptAttachment();
+    if (PlatformInfos.isWeb) {
+      final blob = html.Blob([videoFile.bytes]);
+      if (lastSelectedVideoEventId == event.eventId) {
+        _playVideoActionWeb(context, html.Url.createObjectUrlFromBlob(blob));
+      }
+      return html.Url.createObjectUrlFromBlob(blob);
+    } else {
+      final tempDir = await getTemporaryDirectory();
+      final fileName = Uri.encodeComponent(
+        event.attachmentOrThumbnailMxcUrl()!.pathSegments.last,
+      );
+      final file = File('${tempDir.path}/${fileName}_${videoFile.name}');
+      if (await file.exists() == false && videoFile.bytes != null) {
+        await file.writeAsBytes(videoFile.bytes!);
+      }
+      if (lastSelectedVideoEventId == event.eventId) {
+        _playVideoActionMobile(context, file.path);
+      }
+      return file.path;
+    }
+  }
+
+  String? lastSelectedVideoEventId;
+
+  void _playVideoActionWeb(BuildContext context, String uri) async {
+    await showDialog(
+      context: context,
+      useRootNavigator: true,
+      builder: (context) {
+        return VideoViewerDialog(path: uri);
+      },
+    );
+  }
+
+  void _playVideoActionMobile(BuildContext context, String filePath) async {
+    await showDialog(
+      context: context,
+      useRootNavigator: false,
+      builder: (context) {
+        return VideoViewerDialog(path: filePath);
+      },
+    );
+  }
```

Comment:
```suggestion
  void _playVideoActionWeb(BuildContext context, String uri) async {
    await showDialog(
      context: context,
      useRootNavigator: PlatFormInfo.isWeb,
      builder: (context) {
        return VideoViewerDialog(path: uri);
      },
    );
  }
```

Reply:
the condition is check above already.

Reply:
agree with huy, also need util to reuse it, DRY

Reply:
Any update?

Reply:
remove already, please check an resolve